src/common/utils/deploy.utils.ts

```diff
@@ -76,7 +78,7 @@ export default class DeployUtils {
         component_secrets['*'] = {};
       }
       // Shallow merge to ensure CLI arguments replace anything from the secrets file
-      component_secrets['*'] = { ...component_secrets['*'], ...extra_secrets };
+      component_secrets['*'] = { ...component_secrets['*'], ...extra_secrets } as Dictionary<SecretType>;
```

Comment:
Why do we need the cast here?

Reply:
This is because DeployUtils.getExtraSecrets(individual_secrets) returns `Dictionary<string | number | undefined>` and not `Dictionary<SecretType>`. I will change the casting to `const extra_secrets = DeployUtils.getExtraSecrets(individual_secrets) as Dictionary<SecretType>;` instead.